Automattic/themes
# Blockbase: Simplify the approach to font customization

#### Changes proposed in this Pull Request:
This changes our approach to font customization in Blockbase. The current implementation restricts our division of body and heading settings to specific blocks. This approach means that themes can use heading or body fonts on any blocks and the customization will still work.

This approach is much simpler than our previous implementation. We are using semantic rather than descriptive names for our typography settings (heading/body instead of Times/Arial). This means that in order to edit a font we just need to define a new typography setting with the same key.

This was made more complex by the fact that we have to deal with sites that are using the old settings. In this case we only show the reset button, and then when they reset we allow them the select a font again.

To test:
1. Switch to trunk
2. Select a Blockbase theme
3. Go to the Customizer and select a custom font
4. Confirm that you see the new font in the front end
5. Switch to this branch
6. Go to the Customizer
7. Confirm that you see the reset button, but not your selected fonts
8. Press the reset button
9. Confirm that the preview reloads and shows the default fonts
10. Confirm that the frontend shows the default fonts
11. Select a custom font
12. Confirm that the preview shows the correct font
13. Save changes
14. Confirm that the frontend shows the correct font
